omnibus/files/private-chef-cookbooks/private-chef/templates/default/haproxy.cfg.erb

```diff
@@ -0,0 +1,26 @@
+global
+        stats socket /var/opt/opscode/haproxy/haproxy.sock mode 600
+
+frontend postgresql
+         bind <%= @listen %>:<%= @local_postgresql_port %>
+         mode tcp
+         default_backend chef_backend_postgresql
+
+frontend elasticsearch
+         bind <%= @listen %>:<%= @local_elasticsearch_port %>
+         mode tcp
+         default_backend chef_backend_elasticsearch
+
+backend chef_backend_postgresql
+        mode tcp
+        option httpchk GET /leader HTTP/1.1\r\nHost:localhost:<%= @leaderl_healthcheck_port %>\r\n\r\n
+        <% @chef_backend_members.each do |name, ip| -%>
+        server <%= name %> <%= ip %>:<%= @remote_postgresql_port %> check port <%= @leaderl_healthcheck_port %> rise 1 fall 1
```

Comment:
I think we can specify defaults here up above for all the servers. Like:
```
default-server inter 3s fall 3 rise 2
server name ip:port check port
server name ip:port check port
...
```
https://www.haproxy.com/doc/aloha/7.0/haproxy/healthchecks.html

Reading the docs a little more there's actually built in support for postgres as well:

```
backend be_pgsql
 [...]
 option pgsql-check
 server srv1 10.0.0.1:5432 check
 server srv2 10.0.0.2:5432 check
```